Iterable-iOS-SDK/IterableAPI.m

```diff
@@ -683,6 +683,23 @@ - (void)updateUser:(NSDictionary *)dataFields mergeNestedObjects:(BOOL)mergeNest
     }
 }
 
+// documented in IterableAPI.h
+- (void)updateEmail:(NSString *)newEmail onSuccess:(OnSuccessHandler)onSuccess onFailure:(OnFailureHandler)onFailure
+{
+    if (newEmail) {
+        NSDictionary *args = @{
+                ITBL_KEY_CURRENT_EMAIL: self.email,
+                ITBL_KEY_NEW_EMAIL: newEmail
+        };
+
+        NSURLRequest *request = [self createRequestForAction:ENDPOINT_UPDATE_EMAIL withArgs:args];
+        [self sendRequest:request onSuccess:^(NSDictionary *dictionary) {
+            _email = newEmail;
+            onSuccess(dictionary);
+        } onFailure:onFailure];
+    }
```

Comment:
if you decide not to mark `newEmail` as non nullable, should you fire the `onFailure` handler here?

Reply:
Yeah, that's a good question. onSuccess & onFailure are implicitly non-nullable, but are also not enforced in ObjC, and calling a null block actually crashes the app. I'm not really sure whether it makes sense to add these checks everywhere just because ObjC compiler doesn't enforce this...

Reply:
What do you think of calling our default handler if the onFailure is nil so the app at least gets some feedback rather than silently failing.

Reply:
It seems like the current convention is that we use default handlers for methods without handlers in the signature: if the user explicitly passes nil as a handler (which is only possible in ObjC), we just don't call anything then.